gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/configuration/application/registration/ClientRegistrationServiceImpl.java

```diff
@@ -333,6 +333,23 @@ public ClientRegistrationResponse register(ExecutionContext executionContext, Ne
         return registrationProviderClient.register(clientRegistrationRequest);
     }
 
+    private static ClientRegistrationRequest convert(NewApplicationEntity application) {
+        var mapped = ClientRegistrationMapper.INSTANCE.toClientRegistrationRequest(application);
+        var objectMapper = new ObjectMapper();
+        ObjectNode mappedJsonNode = objectMapper.valueToTree(mapped);
+
+        var additionalClientMetadata = application.getSettings().getOAuthClient().getAdditionalClientMetadata();
+        if (additionalClientMetadata instanceof ObjectNode additionalClientMetadataNode) {
+            additionalClientMetadataNode.setAll(mappedJsonNode);
+            mappedJsonNode = additionalClientMetadataNode;
+        }
+        try {
+            return objectMapper.treeToValue(mappedJsonNode, ClientRegistrationRequest.class);
+        } catch (JsonProcessingException e) {
+            return mapped;
+        }
```

Comment:
to be honest I don't understand why all of this is not part of the `ClientRegistrationMapper` you just created before